src/upgrades/mongodb_upgrade.py

```diff
@@ -289,7 +340,11 @@ def is_cluster_healthy(self) -> bool:
                 logger.error("Cannot proceed with upgrade. Service mongod is not running")
                 return False
 
-        if not self.charm.status.are_all_units_ready_for_upgrade():
+        if not self.charm.status.is_current_unit_ready(
+            ignore_unhealthy_upgrade=True
```

Comment:
Doesn't this wrong the output of `is_cluster_healthy`? given that not all the cluster is effectively healthy. And do we want to proceed with the upgrade in this case?

Reply:
It's actually more nuanced than that - let me add a comment